pcsx2/Frontend/INISettingsInterface.cpp

```diff
@@ -0,0 +1,242 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2021  PCSX2 Dev Team
+ *
+ *  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU Lesser General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PrecompiledHeader.h"
+
+#include "INISettingsInterface.h"
+#include "common/FileSystem.h"
+#include "common/Console.h"
+#include "common/StringUtil.h"
+#include <algorithm>
+#include <iterator>
+
+INISettingsInterface::INISettingsInterface(std::string filename)
+	: m_filename(std::move(filename))
+	, m_ini(true, true)
+{
+}
+
+INISettingsInterface::~INISettingsInterface()
+{
+	if (m_dirty)
+		Save();
+}
+
+bool INISettingsInterface::Load()
+{
+	SI_Error err = SI_FAIL;
+	auto fp = FileSystem::OpenManagedCFile(m_filename.c_str(), "rb");
+	if (fp)
+		err = m_ini.LoadFile(fp.get());
+
+	return (err == SI_OK);
+}
+
+bool INISettingsInterface::Save()
+{
+	SI_Error err = SI_FAIL;
+	std::FILE* fp = FileSystem::OpenCFile(m_filename.c_str(), "wb");
+	if (fp)
+	{
+		err = m_ini.SaveFile(fp, false);
+		std::fclose(fp);
+	}
+
+	if (err != SI_OK)
+	{
+		Console.Warning("Failed to save settings to '%s'.", m_filename.c_str());
+		return false;
+	}
+
+	m_dirty = false;
+	return true;
+}
+
+void INISettingsInterface::Clear()
+{
+	m_ini.Reset();
+}
+
+bool INISettingsInterface::GetIntValue(const char* section, const char* key, int* value) const
+{
+	const char* str_value = m_ini.GetValue(section, key);
```

Comment:
Why is for example GetLongValue inappropriate here?

https://github.com/brofield/simpleini/blob/master/SimpleIni.h#L885

Guessing from the documentation, the default value is a problem because we cannot set a "value not found" value that is guaranteed to always work?